src/Agent/NewRelic/Agent/Core/SharedInterfaces/Environment.cs

```diff
@@ -3,58 +3,94 @@
 
 using System;
 using System.Collections;
+using System.Collections.Concurrent;
 using System.Collections.Generic;
 using System.Linq;
-using NewRelic.Agent.Extensions.Logging;
 
-namespace NewRelic.Agent.Core.SharedInterfaces
+namespace NewRelic.Agent.Core.SharedInterfaces;
+
+/// <summary>
+/// Provides methods for retrieving command-line arguments and environment variables for the current process.
+/// Serves as a wrapper around System.Environment to facilitate testing and abstraction.
+/// </summary>
+/// <remarks>This class offers convenient access to environment-related information, such as command-line
+/// arguments and environment variables, with optional caching for improved performance. It is intended for use in
+/// scenarios where environment data needs to be accessed or filtered within an application. All members are instance
+/// methods; thread safety is ensured for environment variable retrieval.</remarks>
+public class Environment : IEnvironment
 {
-    public class Environment : IEnvironment
+    private readonly ConcurrentDictionary<string, string> _environmentVariableCache = new();
+
+    public string[] GetCommandLineArgs() => System.Environment.GetCommandLineArgs();
+
+    /// <summary>
+    /// Retrieves the value of the specified environment variable from the current process, caching the result for future calls.
+    /// </summary>
+    /// <param name="variable">The name of the environment variable to retrieve. Cannot be null.</param>
+    /// <returns>The value of the environment variable specified by <paramref name="variable"/>; or null if the environment
+    /// variable is not found.</returns>
+    /// <exception cref="ArgumentNullException">Thrown if <paramref name="variable"/> is null.</exception>
+    public string GetEnvironmentVariable(string variable)
     {
-        public string[] GetCommandLineArgs()
-        {
-            return System.Environment.GetCommandLineArgs();
-        }
+        if (variable == null)
+            throw new ArgumentNullException(nameof(variable));
+        return _environmentVariableCache.GetOrAdd(variable, _ => System.Environment.GetEnvironmentVariable(variable));
```

Comment:
You might be able to avoid using a lambda here if did something more like the following.
```suggestion
        return _environmentVariableCache.GetOrAdd(variable, System.Environment.GetEnvironmentVariable);
```
I don't know if the compiler will correctly disambiguate which overload to use though.